modin/pandas/test/test_groupby.py

```diff
@@ -1790,42 +1792,50 @@ def test_not_str_by(by, as_index):
     "agg_func",
     [
         pytest.param(
-            lambda grp: grp.apply(lambda df: df.dtypes), id="modin_dtype_impl"
+            lambda grp: grp.apply(lambda df: df.dtypes), id="modin_dtypes_impl"
         ),
-        pytest.param(lambda grp: grp.apply(lambda df: df.sum()), id="apply(sum)"),
+        pytest.param(lambda grp: grp.apply(lambda df: df.sum()), id="apply_sum"),
         pytest.param(lambda grp: grp.count(), id="count"),
         pytest.param(lambda grp: grp.nunique(), id="nunique"),
+        # Integer key means the index of the column to replace it with.
+        # 0 and -1 are considered to be the indices of the columns to group on.
+        pytest.param({1: "sum", 2: "nunique"}, id="dict_agg_no_intersection_with_by"),
         pytest.param(
-            {"c": "sum", "d": "nunique"}, id="dict_agg_no_intersection_with_by"
-        ),
-        pytest.param(
-            {"b": "mean", "c": "sum", "d": "nunique"},
+            {1: "sum", 2: "nunique", -1: "mean"},
             id="dict_agg_has_intersection_with_by",
         ),
         pytest.param(
-            {"a": "nunique", "c": "sum", "d": "nunique"},
+            {0: "nunique", 1: "sum", 2: "nunique"},
             id="dict_agg_has_intersection_with_categorical_by",
         ),
     ],
 )
+# There are two versions of the `handle_as_index` method: the one accepting pandas.DataFrame from
+# the execution kernel and backend agnostic. This parameter indicates which one implementation to use.
+@pytest.mark.parametrize("use_backend_agnostic_method", [True, False])
 def test_handle_as_index(
-    internal_by_length, external_by_length, has_categorical_by, agg_func, request
+    internal_by_length,
+    external_by_length,
+    has_categorical_by,
+    agg_func,
+    use_backend_agnostic_method,
+    request,
 ):
     """
-    Test ``GroupBy.handle_as_index``.
+    Test ``modin.core.dataframe.algebra.default2pandas.groupby.GroupBy.handle_as_index``.
 
     The role of the ``handle_as_index`` method is to build a groupby result considering
     ``as_index=False`` from the result that was computed with ``as_index=True``.
 
     So the testing flow is the following:
         1. Compute GroupBy result with the ``as_index=True`` parameter via Modin.
         2. Build ``as_index=False`` result from the ``as_index=True`` using ``handle_as_index`` method.
-        3. Compute GroupBy result with the ``as_index=False`` parameter via Pandas (is the reference result).
+        3. Compute GroupBy result with the ``as_index=False`` parameter via pandas as the reference result.
         4. Compare the result from the second step with the reference.
     """
     by_length = sum((internal_by_length, external_by_length))
```

Comment:
```suggestion
    by_length = internal_by_length + external_by_length
```
isn't it the same, huh?